src/data/zmod/algebra.lean

```diff
@@ -15,6 +15,18 @@ namespace zmod
 
 variables (R : Type*) [ring R]
 
+instance (p : ℕ) : subsingleton (algebra (zmod p) R) :=
+⟨λ x y, begin
+  casesI p,
+  { exact @@subsingleton.elim int_algebra_subsingleton x y },
+  refine x.algebra_ext y (λ r, _),
+  convert_to (@algebra_map _ _ _ _ x) ((coe : zmod p.succ → zmod p.succ) r) =
+             (@algebra_map _ _ _ _ y) ((coe : zmod p.succ → zmod p.succ) r),
+  { rw [cast_id] },
+  { rw [cast_id] },
+  rw [←nat_cast_val r, ←nat.smul_one_eq_coe, map_nsmul, map_nsmul, map_one, map_one]
+end⟩
```

Comment:
```suggestion
⟨λ x y, algebra.algebra_ext _ _ $ ring_hom.congr_fun $ subsingleton.elim _ _⟩
```

Reply:
Arguably `algebra_ext` is stated badly and the `ring_hom.congr_fun` shouldn't even be needed